### PR TITLE
Develop/rose atc

### PR DESCRIPTION
**Pull request by Whitney-Rene**
_rose: a therapy companion simplifies the process of keeping a therapy journal and provides a convenient place for me to record these moments – the "Rose" for my successes, the "Thorn" for my areas of improvement, the "Bud" for ideas or topics I'd like to discuss in the upcoming session, and the "Homework" for tasks assigned to me._

**The current status of the pushed up files:**
I have initialized 5 components.  Each component has basic set up with simple h1/p/img tags and comments for future plans.

I installed react-router and imported the necessary components (browserrouter, route, routes) into app.jsx.  I added a few simple routes, so that the user can navigate between home and other pages.

There are 2 utils folders, one in client and one in server.  I have added a couple functions I hope to import into other components in the future.  It is quite messy with comments.  I will clean it up later.

I've added comments to server.js with future plans. I will clear out the comment later.

### CODE REVIEW

File: client/src/App.jsx [hunk 1]
```diff
@@ -1,35 +1,34 @@
 import { useState, useEffect } from 'react';
+//'switch' has been replaced with 'routes'
```
Comment (edDoganFreenome) [inline client/src/App.jsx]:
```suggestion
//'Switch' has been replaced with 'Routes'
```
Even in the comments, paying attention to the capitalization will help you in the future.

File: client/src/components/CreateRoseBudThorn.jsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    <div classname="create" >
+
+    <h2>Create a new RoseBudThorn</h2>
+
+    </div>
+  )
+}
+
+//learned:
+//sfc tab
```
Comment (edDoganFreenome) [inline client/src/components/CreateRoseBudThorn.jsx]: sfc tab?

File: client/src/components/HomePage.jsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  return (
+    <div>
+        <br />
+
+        <img 
+        src="https://i.etsystatic.com/9472596/r/il/749ea1/3821016144/il_1588xN.3821016144_csuf.jpg" 
+        alt='bouquet of flowers' 
+        style={{ width: '200px', height: 'auto' }}
+        />
```
Comment (edDoganFreenome) [inline client/src/components/HomePage.jsx]:
```suggestion
        <img 
            src="https://i.etsystatic.com/9472596/r/il/749ea1/3821016144/il_1588xN.3821016144_csuf.jpg" 
            alt='bouquet of flowers' 
            style={{ width: '200px', height: 'auto' }}
        />
```

Nit comment, but it will help your overall experience with your code editor too. 
For this project, I'd install `prettier` code formatter. You probably don't need to change the settings but turn on the 'format on save' functionality so that it automatically adjusts your code/indentation etc. automatically. It also helps your reviewers :).

File: client/src/components/HomePage.jsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import ListLatestEntries from './ListLatestEntries'
+
+export default function HomePage() {
+
+    //fetch async function to db where entries are
+
+  return (
+    <div>
+        <br />
```
Comment (edDoganFreenome) [inline client/src/components/HomePage.jsx]: I'd use `<br />`s in HTML sparingly. Most of the `<br/>`s should be handled with `margin` on your elements. 
This one can be a `margin-top` in the `img` element below. It gives you more flexibility and more control over your design.

File: client/utils/functions.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+
+//my intent is to use this function in the homepage comp of app.jsx, I'd like to house it here to make my code cleaner
+//I want to learn how to possibly resue line 8? can the http be saved as a variable and I add diff endpoints based my server?
```
Comment (edDoganFreenome) [inline client/utils/functions.js]: This is a great idea.

File: server/server.js [hunk 6]
```diff
@@ -16,8 +16,30 @@ app.use(express.json());
...
-})
+});
+
+app.get('/users', async (req, res) => {
+    try {
+        const { rows: users } = await db.query('SELECT * FROM users');
+        res.send(users);
+    } catch (error) {
+        console.error('Error in Database Query:', error);
+        return res.status(400).json({ error });
```
Comment (edDoganFreenome) [inline server/server.js]: I'm not sure about returning a 400 error all the time here. It can be 400 (client side, there's a problem with your request), or 500 (your request was fine, but there's a problem with the server). Maybe an optimization and further reading for later.

File: server/package.json [hunk 7]
```diff
@@ -11,7 +11,6 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
-    "bcrypt": "^5.1.1",
```
Comment (edDoganFreenome) [inline server/package.json:14]: Why is this removed? I see that you're using it in the `hashPasswordUtils` file.

**Review by edDoganFreenome — APPROVED** [review]
Great start to an exciting project!

File: client/src/components/CreateRoseBudThorn.jsx [hunk 8]
```diff
@@ -0,0 +1,13 @@
+
+export default function CreateRoseBudThorn() {
+  return (
+    <div classname="create" >
```
Comment (edDoganFreenome) [inline client/src/components/CreateRoseBudThorn.jsx]:
```suggestion
    <div className="create" >
```

Comment (edDoganFreenome) [reply client/src/components/CreateRoseBudThorn.jsx]: Another package suggestion eslint 
https://www.npmjs.com/package/eslint

It will warn you of this kind of typos.

File: client/src/App.jsx [hunk 9]
```diff
@@ -1,35 +1,34 @@
...
-    } catch{
-      console.error('An error occured:', error);
-    }
-  }
+  return (
 
-  useEffect(() => {
-    callBackEnd();
-  }, []);
+      <Router>
```
Comment (plee-fn) [inline client/src/App.jsx]: people usually put BrowserRouter wrapper in ./main.jsx

File: client/src/App.jsx [hunk 10]
```diff
@@ -1,35 +1,34 @@
 import { useState, useEffect } from 'react';
+//'switch' has been replaced with 'routes'
+import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
```
Comment (plee-fn) [inline client/src/App.jsx]:
```suggestion
import { Route, Routes } from 'react-router-dom';
```
move `import { BrowserRouter as Router } from 'react-router-dom'` to your `./main.jsx`

File: client/src/App.jsx [hunk 11]
```diff
@@ -1,35 +1,34 @@
...
+
+import CreateRoseBudThorn from './components/CreateRoseBudThorn';
+import HomePage from './components/HomePage';
+import ListLatestEntrie from './components/ListLatestEntries';
```
Comment (plee-fn) [inline client/src/App.jsx]:
```suggestion
import ListLatestEntries from './components/ListLatestEntries';
```
this may mean you need to update your return statement where you are using this component

File: client/src/App.jsx [hunk 12]
```diff
@@ -1,35 +1,34 @@
...
+import RequestBouquet from './components/RequestBouquet';
+import NavBar from './components/NavBar';
+
```
Comment (plee-fn) [inline client/src/App.jsx]: probably don't need this \n

File: client/src/components/CreateRoseBudThorn.jsx (hunk 8 above)
Comment (plee-fn) [reply client/src/components/CreateRoseBudThorn.jsx]: people also pair eslint with prettier to make everything pretty
https://www.npmjs.com/package/prettier

File: server/db/db-connection.js [hunk 13]
```diff
@@ -1,7 +1,10 @@
...
+console.log(process.env.DB_URI)
+
+const db = new Pool ({
     connectionString: process.env.DB_URI
-})
+});
 
-export { db }; 
\ No newline at end of file
+export default db;
```
Comment (plee-fn) [inline server/db/db-connection.js]: your missing a \n at the end of the file

File: server/server.js [hunk 14]
```diff
@@ -16,8 +16,30 @@ app.use(express.json());
...
+    }
+});
+
+app.get('/entries', async (req, res) => {
+    try {
+        const { rows: entries } = await db.query('SELECT * FROM entries');
+        res.send(entries);
+    } catch (error) {
+        console.error('Error in Database Query:', error);
+        return res.status(400).json({ error });
```
Comment (plee-fn) [inline server/server.js]: same comment from ed above (https://github.com/Whitney-Rene/rose_aTC/pull/1/files#r1373426827)

File: server/utils/hashPasswordUtils.js [hunk 15]
```diff
@@ -1,8 +1,9 @@
 import bcrypt from 'bcrypt';
 
+//my intent is to eventually hash the password of users before inserting in the db, need to think through wherr to import this function
 function hashPassword (password) {
     const saltRounds =10;
     return bcrypt.hashSync(password, saltRounds);
 }
 
-export { hashPassword };
+export { hashPassword };
```
Comment (plee-fn) [inline server/utils/hashPasswordUtils.js]: missing \n at the end of the file

File: client/src/App.jsx (hunk 1 above)
Comment (Whitney-Rene) [reply client/src/App.jsx]: Okay, thank you for bringing this to my attention.

File: client/src/components/CreateRoseBudThorn.jsx (hunk 2 above)
Comment (Whitney-Rene) [reply client/src/components/CreateRoseBudThorn.jsx]: In VScode, if you create a jsx file/new component, instead of manually typing up the basic code to set up a new component, you can type "sfc" in the blank jsx file and hit tab.  This will populate the component with the basic set up.  It is a cool trick I learned.   I will delete it for now, and write in in a "shortcuts" note.
sfc stands for stateless functional component.

File: client/src/components/HomePage.jsx (hunk 3 above)
Comment (Whitney-Rene) [reply client/src/components/HomePage.jsx]: Thank you!  I installed prettier!

File: client/src/components/HomePage.jsx (hunk 4 above)
Comment (Whitney-Rene) [reply client/src/components/HomePage.jsx]: great insight. thank you for helping me make this connection.

File: client/utils/functions.js (hunk 5 above)
Comment (Whitney-Rene) [reply client/utils/functions.js]: Thank you!  I think I have my functions.js and variables.js working.  I'll share in our next mentor meeting!

File: server/server.js (hunk 6 above)
Comment (Whitney-Rene) [reply server/server.js]: Hmm...this is very insightful.  Thank you for sharing.  I changed them to 500, and made a not in the client side to use 400 there for errors.

File: server/package.json (hunk 7 above)
Comment (Whitney-Rene) [reply server/package.json:14]: Philip and I were wondering the same thing!  I do not even remember doing that.  I will put it back.

File: client/src/components/CreateRoseBudThorn.jsx (hunk 8 above)
Comment (Whitney-Rene) [reply client/src/components/CreateRoseBudThorn.jsx]: thank you both!  i tried adding prettier and also researched the web, but could not get it to work.  I'll keep working on it!
I changed the typo.

File: client/src/App.jsx (hunk 9 above)
Comment (Whitney-Rene) [reply client/src/App.jsx]: thank you!  I changed it.

File: client/src/App.jsx (hunk 10 above)
Comment (Whitney-Rene) [reply client/src/App.jsx]: Done!

File: client/src/App.jsx (hunk 12 above)
Comment (Whitney-Rene) [reply client/src/App.jsx]: Done!

File: client/src/App.jsx (hunk 11 above)
Comment (Whitney-Rene) [reply client/src/App.jsx]: Thank you!  Good catch!

File: server/server.js (hunk 14 above)
Comment (Whitney-Rene) [reply server/server.js]: Thank you!

**Comment by Whitney-Rene** [comment]
Thank you for your thoughtful comments and suggestions!
